src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -3051,6 +3110,33 @@ bool TheoryStringsRewriter::checkEntailArithInternal(Node a)
   return false;
 }
 
+Node TheoryStringsRewriter::decomposeSubstrChain(Node s,
+                                                 std::vector<Node>& ss,
+                                                 std::vector<Node>& ls)
+{
```

Comment:
Assert that vectors are empty.

Reply:
This may be too strict, e.g. here's a hypothetical piece of code:

Node curr = ...;
std::vector< Node > ss;
std::vector< Node > ls;
while( curr.getKind()==SUBSTR )
{
  Node x = decomposeSubstr( n, ss, ls );
  curr = getSubstrEqualTo( x );  
}

where we'd want to use decomposeSubstr to accumulate the chain on multiple iterations.

Reply:
I added these assertions.